auth_backends/tests/test_pipeline.py

```diff
@@ -58,19 +60,25 @@ def test_update_email(self, mock_set_attribute):
         self.assertEqual(updated_user.email, updated_email)
         self.assertNotEqual(updated_user.email, initial_email)
 
-        mock_set_attribute.assert_called_with('update_email.username_mismatch', False)
+        mock_set_attribute.assert_any_call('update_email.username_mismatch', False)
+        mock_set_attribute.assert_any_call('update_email.rollout_toggle_enabled', False)
+        mock_set_attribute.assert_any_call('update_email.email_updated', True)
 
+    @patch('auth_backends.pipeline.SKIP_UPDATE_EMAIL_ON_USERNAME_MISMATCH.is_enabled')
     @patch('auth_backends.pipeline.set_custom_attribute')
-    def test_update_email_with_none(self, mock_set_attribute):
+    def test_update_email_with_none(self, mock_set_attribute, mock_toggle):
         """ Verify that user email is not updated if email value is None. """
+        mock_toggle.return_value = False
         old_email = self.user.email
 
         update_email(self.strategy, {'email': None, 'username': 'test_user'}, user=self.user)
 
         updated_user = User.objects.get(pk=self.user.pk)
         self.assertEqual(updated_user.email, old_email)
 
-        mock_set_attribute.assert_called_with('update_email.username_mismatch', False)
+        mock_set_attribute.assert_any_call('update_email.username_mismatch', False)
+        mock_set_attribute.assert_any_call('update_email.rollout_toggle_enabled', False)
+        # email_updated should not be called since email wasn't updated
```

Comment:
Nit: The mock has an array of calls args. Consider checking that the custom attribute was not called in place of a comment. Consider adding the same check to the True case to show that the code is appropriately checking, because it would be simple to code up a false positive that isn't really checking what you want.